fs/base.py

```diff
@@ -1219,17 +1220,20 @@ def removetree(self, dir_path):
             dir_path (str): Path to a directory on the filesystem.
 
         """
-        _dir_path = abspath(normpath(dir_path))
         with self._lock:
-            walker = walk.Walker(search="depth")
-            gen_info = walker.info(self, _dir_path)
-            for _path, info in gen_info:
-                if info.is_dir:
-                    self.removedir(_path)
-                else:
-                    self.remove(_path)
-            if _dir_path != "/":
-                self.removedir(dir_path)
+            if self.hassyspath(path):
+                shutil.rmtree(self.getsyspath(path))
```

Comment:
`shutil.rmtree` can potentially raise `OSError`, we would want to wrap it using the `onerror` callback, probably turning it into an `fs.errors.OperationFailed`.